README.md

```diff
@@ -1 +1,198 @@
+  </plugin>
+</gazebo>
+```
+
+#### Set up controllers
+
+Use the tag `<parameters>` inside `<plugin>` to set the YAML file with the controller configuration.
+
+```xml
+<gazebo>
+  <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
+    <parameters>$(find gazebo_ros2_control_demos)/config/cartpole_controller.yaml</parameters>
+  </plugin>
+<gazebo>
+```
+
+This controller publishes the state of all resources registered to a
+`hardware_interface::JointStateInterface` to a topic of type `sensor_msgs/JointState`. The following is a basic configuration of the controller.
+
+```yaml
+joint_state_controller:
+  ros__parameters:
+    type: joint_state_controller/JointStateController
+    publish_rate: 50
+```
+
+This controller creates an action called `/cart_pole_controller/follow_joint_trajectory` of type `control_msgs::action::FollowJointTrajectory`.
+
+```yaml
+cart_pole_controller:
+  ros__parameters:
+    type: joint_trajectory_controller/JointTrajectoryController
+    joints:
+       - slider_to_cart
+    write_op_modes:
+       - slider_to_cart
+    state_publish_rate: 25
+    action_monitor_rate: 20
+    constraints:
+      stopped_velocity_tolerance: 0.05
+      goal_time: 5
+```
+
+#### Setting PID gains
+
+To set the PID gains for a specific joint you need to define them inside `<plugin><ros></plugin></ros>`. Using the generic way of defining parameters with `gazebo_ros`. The name of the parameter correspond the name of the joint followed by a dot and the name of the parameter: `p`, `i`, `d`, `i_clamp_max`, `i_clamp_min` and `antiwindup`.
+
+```xml
+<gazebo>
+  <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
+    <ros>
+      <namespace>/</namespace>
+      <parameter name="slider_to_cart.p" type="double">50.0</parameter>
+      <parameter name="slider_to_cart.i" type="double">10.0</parameter>
+      <parameter name="slider_to_cart.d" type="double">15.0</parameter>
+      <parameter name="slider_to_cart.i_clamp_max" type="double">3.0</parameter>
+      <parameter name="slider_to_cart.i_clamp_min" type="double">-3.0</parameter>
+      <parameter name="slider_to_cart.antiwindup" type="bool">false</parameter>
+    </ros>
+    ...
+  </plugins>
+</gazebo>
+```
+
+#### Executing the examples
+
+There are some examples in the `gazebo_ros2_control_demos` package. These examples allow to launch a cart in a 30 meter rail.
+
+![](img/cart.gif)
+
+You can run some of the configuration running the following commands:
+
+```bash
+ros2 launch gazebo_ros2_control_demos cart_example_position_pid.launch.py
+ros2 launch gazebo_ros2_control_demos cart_example_position.launch.py
+ros2 launch gazebo_ros2_control_demos cart_example_velocity.launch.py
+```
+
+Send example commands:
+
+When the Gazebo world is launched you can run some of the following commads to move the cart.
+
+```bash
+ros2 run gazebo_ros2_control_demos example_position
+ros2 run gazebo_ros2_control_demos example_velocity
```

Comment:
I tried a few combinations of these commands with the launch files above and I'm always getting:

```
node created
terminate called after throwing an instance of 'std::runtime_error'
  what():  could not get action server
```

Reply:
probably Gazebo is not launched properly

Reply:
They work now